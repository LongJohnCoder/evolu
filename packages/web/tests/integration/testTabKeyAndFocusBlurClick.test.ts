import { pageDom, pageGoto } from './helpers';

beforeEach(async () => {
  await pageGoto('testTabKeyAndFocusBlurClick');
});

test('render', async () => {
  await expect(await pageDom()).toMatchSnapshot();
});

test('focus blur via tab, tab, tab', async () => {
  await page.keyboard.press('Tab');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();

  await page.keyboard.press('Tab');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();

  await page.keyboard.press('Tab');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();
});

test('click to focus button, click to blur button', async () => {
  await page.click('button.focus');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();

  await page.click('button.blur');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();
});
