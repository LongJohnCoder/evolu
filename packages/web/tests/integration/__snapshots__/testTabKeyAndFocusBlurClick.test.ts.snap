// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click to focus button, click to blur button 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·heading·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    class="jsx-442929638"
  >
    ·Schema Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-vlbfxt-heading"
      >
        ·heading·
      </div>
      <div
        class="css-1yjna00-paragraph"
      >
        ·paragraph·
      </div>
      <div
        class="css-1n9mwyn-list"
      >
        <div
          class="css-0"
        >
          ·listitem·
        </div>
      </div>
      <img
        alt="Square placeholder image 80px"
        height="80"
        src="https://via.placeholder.com/80"
        width="80"
      />
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    style="margin-bottom:24px"
  >
    <button
      class="focus"
      disabled=""
      tabindex="0"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      tabindex="-1"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`click to focus button, click to blur button 2`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·heading·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    class="jsx-442929638"
  >
    ·Schema Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-vlbfxt-heading"
      >
        ·heading·
      </div>
      <div
        class="css-1yjna00-paragraph"
      >
        ·paragraph·
      </div>
      <div
        class="css-1n9mwyn-list"
      >
        <div
          class="css-0"
        >
          ·listitem·
        </div>
      </div>
      <img
        alt="Square placeholder image 80px"
        height="80"
        src="https://via.placeholder.com/80"
        width="80"
      />
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·3·
      · ·
      ·{"hasFocus":false,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    style="margin-bottom:24px"
  >
    <button
      class="focus"
      tabindex="-1"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      tabindex="0"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`focus blur via tab, tab, tab 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·heading·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true}·
    </span>
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    class="jsx-442929638"
  >
    ·Schema Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-vlbfxt-heading"
      >
        ·heading·
      </div>
      <div
        class="css-1yjna00-paragraph"
      >
        ·paragraph·
      </div>
      <div
        class="css-1n9mwyn-list"
      >
        <div
          class="css-0"
        >
          ·listitem·
        </div>
      </div>
      <img
        alt="Square placeholder image 80px"
        height="80"
        src="https://via.placeholder.com/80"
        width="80"
      />
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    style="margin-bottom:24px"
  >
    <button
      class="focus"
      tabindex="-1"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      tabindex="0"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`focus blur via tab, tab, tab 2`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·heading·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·3·
      · ·
      ·{"hasFocus":false,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true}·
    </span>
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    class="jsx-442929638"
  >
    ·Schema Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-vlbfxt-heading"
      >
        ·heading·
      </div>
      <div
        class="css-1yjna00-paragraph"
      >
        ·paragraph·
      </div>
      <div
        class="css-1n9mwyn-list"
      >
        <div
          class="css-0"
        >
          ·listitem·
        </div>
      </div>
      <img
        alt="Square placeholder image 80px"
        height="80"
        src="https://via.placeholder.com/80"
        width="80"
      />
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    style="margin-bottom:24px"
  >
    <button
      class="focus"
      disabled=""
      tabindex="0"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      tabindex="-1"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`focus blur via tab, tab, tab 3`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·heading·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·3·
      · ·
      ·{"hasFocus":false,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true}·
    </span>
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    class="jsx-442929638"
  >
    ·Schema Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-vlbfxt-heading"
      >
        ·heading·
      </div>
      <div
        class="css-1yjna00-paragraph"
      >
        ·paragraph·
      </div>
      <div
        class="css-1n9mwyn-list"
      >
        <div
          class="css-0"
        >
          ·listitem·
        </div>
      </div>
      <img
        alt="Square placeholder image 80px"
        height="80"
        src="https://via.placeholder.com/80"
        width="80"
      />
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·3·
      · ·
      ·{"hasFocus":false,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true}·
    </span>
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    style="margin-bottom:24px"
  >
    <button
      class="focus"
      tabindex="-1"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      tabindex="0"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;

exports[`render 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size: 24px;"
      >
        ·heading·
      </div>
      <div
        style="font-size: 16px;"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    class="jsx-442929638"
  >
    ·Schema Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-vlbfxt-heading"
      >
        ·heading·
      </div>
      <div
        class="css-1yjna00-paragraph"
      >
        ·paragraph·
      </div>
      <div
        class="css-1n9mwyn-list"
      >
        <div
          class="css-0"
        >
          ·listitem·
        </div>
      </div>
      <img
        alt="Square placeholder image 80px"
        height="80"
        src="https://via.placeholder.com/80"
        width="80"
      />
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-378398407"
      title="{'type':'document','children':[{'type':'heading','children':[{'text':'heading'}]},{'type':'paragraph','children':[{'text':'paragraph'}]},{'type':'list','children':[{'type':'listitem','children':[{'text':'listitem'}]}]},{'type':'image','src':'https://via.placeholder.com/80','alt':'Square placeholder image 80px','width':80,'height':80,'children':[]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false}·
    </span>
  </pre>
  <div
    style="margin-bottom:24px"
  >
    <button
      class="focus"
      tabindex="-1"
      type="button"
    >
      ·focus·
    </button>
    <button
      class="blur"
      disabled=""
      tabindex="0"
      type="button"
    >
      ·blur·
    </button>
  </div>
</div>
`;
